libsolutil/Common.h

```diff
@@ -133,4 +133,26 @@ class ScopeGuard
 	std::function<void(void)> m_f;
 };
 
+/// RAII utility class that sets the value of a variable for the current scope and restores it to its old value
+/// during its destructor.
+template<typename V>
+class SetValueForCurrentScope
+{
+public:
+	explicit SetValueForCurrentScope(V& _variable, V&& _value): m_variable(_variable), m_oldValue(std::move(_value))
```

Comment:
Would this even work?
```suggestion
	explicit SetValueForCurrentScope(V& _variable, V&& _value = {}): m_variable(_variable), m_oldValue(std::move(_value))
```

Reply:
If not exactly like that it'd at least be possible to allow not specifying a value *some* other way.
But it depends a bit on how we name this class in the end
``SetValueForCurrentScope guard(var,  {});`` seems more expressive to me than ``SetValueForCurrentScope guard(var);``...

I did consider
```
	template<typename... Args>
	explicit SetValueForCurrentScope(V& _variable, Args&& _args): m_variable(_variable), m_oldValue(std::forward<Args>(_args)...)
```
which would in-place construct the new value - but using that ``SetValueForCurrentScope guard(var,  {});`` doesn't work anymore, since it cannot deduce the type of ``{}``. But if we rename things, maybe that's best...

Reply:
So I renamed, but didn't add support for a default value for now, since it might suggest that the to-be-saved-and-restored value remains unchanged, while simple `{}` as second constructor argument don't hurt too much, but if you want, I can still change it.